### PR TITLE
🥼🧪✨ Part 2 - Documentation Cleanup

### PR DESCRIPTION
**Pull request by keyfer**
## Description

This PR is a followup to https://github.com/Shopify/web-configs/pull/184.

In this PR we:

- [x] Remove documentation not suited for this open-source repository
- [x] Update the PR template
- [x] Update the new issue templates
- [x] Cleaned up `.gitignore`
- [x] Include BundlePhobia on Readme
- [x] OpenSource our contributions guideline, and include it on the Readme
- [x] Add CLA bot to the repository

The documentation regarding internal documents, handbooks, and learnings will be updated and compiled for our internal developer handbooks. I've kept copies of each of them, so they won't be lost during this migration.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -38,6 +38,9 @@ lerna-debug.log
 # IntelliJ IDE ignore
 .idea
 
+# Nova ignore
```
Comment (markrcote) [inline .gitignore]: I would omit this, as per the gitignore man page:

> Patterns which a user wants Git to ignore in all situations (e.g., backup or temporary files generated by the user's editor of choice) generally go into a file specified by core.excludesFile in the user's ~/.gitconfig.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,19 +1,15 @@
 ## Description
 
-Fixes (issue #)
```
Comment (gmcgibbon) [inline .github/PULL_REQUEST_TEMPLATE.md:3]: Personally, I forget to link issues if I don't see this.

File: .gitignore [hunk 3]
```diff
@@ -1,48 +1,7 @@
-# Built style type definitions
```
Comment (keyfer) [inline .gitignore:1]: These were all duplicated for some reason, this is just removing those duplicates.

File: .gitignore [hunk 4]
```diff
@@ -35,9 +38,6 @@ lerna-debug.log
 # Debug logs from yarn
 **/yarn-error.log
 
-# IntelliJ IDE ignore
-.idea
-
 .DS_Store
```
Comment (gmcgibbon) [inline .gitignore:41]: I think you can remove this too, OSX users should know to ignore this in their dotfiles.

File: README.md [hunk 5]
```diff
@@ -5,28 +5,71 @@
...
+| [stylelint-plugin](packages/stylelint-plugin) | [![npm version](https://badge.fury.io/js/%40shopify%2Fstylelint-plugin.svg)](https://badge.fury.io/js/%40shopify%2Fstylelint-plugin) | [![npm bundle size (minified + gzip)](https://img.shields.io/bundlephobia/minzip/@shopify/stylelint-plugin.svg)](https://img.shields.io/bundlephobia/minzip/@shopify/stylelint-plugin.svg) |
+| [typescript-configs](packages/typescript-configs) | [![npm version](https://badge.fury.io/js/%40shopify%2Ftypescript-configs.svg)](https://badge.fury.io/js/%40shopify%2Ftypescript-configs) | [![npm bundle size (minified + gzip)](https://img.shields.io/bundlephobia/minzip/@shopify/typescript-configs.svg)](https://img.shields.io/bundlephobia/minzip/@shopify/typescript-configs.svg) |
+
+## Contribution
+
+Shopify has adopted a Code of Conduct that we expect contributors to adhere to. Please read the [full text](./CODE_OF_CONDUCT.md) so that you can understand what actions will and will not be tolerated.
```
Comment (gmcgibbon) [inline README.md:29]: CLA bot?

File: README.md [hunk 6]
```diff
@@ -5,28 +5,71 @@
...
+dev clone web-configs
+dev up
+
+# External Contributor - To start working on the codebase, first fork the repo, then clone it
+git clone git@github.com:{your-username}/web-configs.git # replace {your-username} with your Github Handle
+yarn # install project dependencies
+yarn lerna bootstrap
+```
+
+##### What is `dev`
```
Comment (gmcgibbon) [inline README.md:54]:
```suggestion
##### What is `dev`?
```

File: README.md [hunk 7]
```diff
@@ -5,28 +5,71 @@
...
+
+`dev` is a tool to standardize a small set of common tasks across all projects at Shopify. If you're wondering how to execute the dev <cmd> commands as an external contributor, you can see where they are defined in this project's `dev.yml` file.
+
+#### Documentation
+
+If your change affects the public API of any packages within this repository (i.e. adding or changing arguments to a function, adding a new function, changing the return value, etc), please ensure the documentation is also updated to reflect this. Documentation is in the `README.md` files of each package. If further documentation is needed please communicate via a Github Issue.
```
Comment (gmcgibbon) [inline README.md:60]:
```suggestion
If your change affects the public API of any packages within this repository (i.e. adding or changing arguments to a function, adding a new function, changing the return value, etc), please ensure the documentation is updated, and a changelog is added to reflect this. Documentation is in the `README.md` files of each package. If further documentation is needed please communicate via a Github Issue.
```

File: README.md [hunk 8]
```diff
@@ -5,28 +5,71 @@
...
+
+#### Testing
+
+The packages in this repository are used in mission-critical production scenarios. As such, we do not merge any untested code. 
+
+To run the full test suite, simply run `dev test` or `yarnpkg test`.
```
Comment (gmcgibbon) [inline README.md:66]:
```suggestion
To run the full test suite, simply run `dev test` or `yarn test`.
```
Less characters, is there a difference?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 9]
```diff
@@ -22,4 +18,4 @@ Please delete options that are not relevant.
 
 ## Checklist
 
-- [ ] I have added a changelog entry, prefixed by the type of change noted above (Documentation fix and Test update does not need a changelog as we do not publish new version)
+- [ ] I have added a changelog entry, prefixed by the type of change noted above (Documentation fix and Test update does not need a changelog as we do not publish new version for these changes)
```
Comment (markrcote) [inline .github/PULL_REQUEST_TEMPLATE.md:21]: "a new version"?

File: README.md (hunk 7 above)
Comment (markrcote) [inline README.md:60]: "GitHub"

File: templates/ROOT_README.hbs.md [hunk 10]
```diff
@@ -5,23 +5,66 @@
...
+yarn lerna bootstrap
+```
+
+##### What is `dev`
+
+`dev` is a tool to standardize a small set of common tasks across all projects at Shopify. If you're wondering how to execute the dev <cmd> commands as an external contributor, you can see where they are defined in this project's `dev.yml` file.
+
+#### Documentation
+
+If your change affects the public API of any packages within this repository (i.e. adding or changing arguments to a function, adding a new function, changing the return value, etc), please ensure the documentation is also updated to reflect this. Documentation is in the `README.md` files of each package. If further documentation is needed please communicate via a Github Issue.
```
Comment (markrcote) [inline templates/ROOT_README.hbs.md:55]: "GitHub"

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (keyfer) [reply .github/PULL_REQUEST_TEMPLATE.md:3]: I actually add this to the commented block below.

**Review by gmcgibbon — APPROVED** [review]

**Review by TheMallen — APPROVED** [review]
Seems good!

**Pull request merged**